maven-core/src/main/java/org/apache/maven/toolchain/java/JavaToolchainImpl.java

```diff
@@ -53,7 +53,7 @@ public String toString() {
     }
 
     public String findTool(String toolName) {
-        File toRet = findTool(toolName, new File(FileUtils.normalize(getJavaHome())));
+        File toRet = findTool(toolName, Paths.get(getJavaHome()).normalize().toFile());
```

Comment:
Use Path.toAbsolutePath().toString() below and don't convert to a File here

Reply:
Fixed